packages/langium/src/parser/langium-parser.ts

```diff
@@ -59,6 +59,7 @@ export class LangiumParser {
     private stack: any[] = [];
     private mainRule!: RuleResult;
     private assignmentMap = new Map<AbstractElement, AssignmentElement | undefined>();
+    private _unorderedGroups: Map<string, boolean[]> = new Map<string, boolean[]>();
```

Comment:
Suggestion: Using a `MultiMap` here makes handling of unset values easier in the parser builder.

```suggestion
    private _unorderedGroups = new MultiMap<string, boolean>();
```

Reply:
Will check it out! Is there a BiMap by the way?